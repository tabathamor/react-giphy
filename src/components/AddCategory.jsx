import { useState } from 'react'

function AddCategory({ onNewCategory }) {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }
    const onSubmit = (e) => {
         e.preventDefault();
         const newInput = inputValue.trim();
         if(newInput.length <= 1) return;
        //  setCategories( (categories) => [ ...categories, inputValue]);
        setInputValue('');

        onNewCategory(newInput)
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder='Buscar gif'
                value={ inputValue }
                onChange={ onInputChange } />
        </form>

    )
}

export default AddCategory