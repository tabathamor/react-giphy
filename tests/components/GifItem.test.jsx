import { render, screen } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

    describe('Pruebas <GifItem />', () => {


        const title = 'title';
        const url = 'https://media.giphy.com/media/l0000001';
        
    test('match a snapshot', async () => {

       const {container} = render(<GifItem title={ title } url={ url } />);
       expect(container).toMatchSnapshot();

    });

    test('Compare title and url', async () => {

        render(<GifItem title={ title } url={ url } />);
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe( url );
        expect(alt).toBe( alt );

    });

    test('Debe mostrar el titulo en el componente', async () => {

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    })
});