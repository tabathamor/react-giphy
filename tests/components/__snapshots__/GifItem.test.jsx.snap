// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem /> match a snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://media.giphy.com/media/l0000001"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
